demos/asp-net-core/ITOps.ViewModelComposition/ISubscribeToCompositionEvents.cs

```diff
@@ -1,10 +1,7 @@
-﻿using Microsoft.AspNetCore.Http;
-using Microsoft.AspNetCore.Routing;
-
-namespace ITOps.ViewModelComposition
+﻿namespace ITOps.ViewModelComposition
 {
     public interface ISubscribeToCompositionEvents : IRouteInterceptor
     {
-        void Subscribe(ISubscriptionStorage subscriptionStorage, RouteData routeData, IQueryCollection query);
+        void Subscribe(IPublishCompositionEvents publisher);
```

Comment:
the removal of `RouteData` and, especially `IQueryCollection`, forces subscribers to always subscribe and if not interested discard the event when it's published. Not a big deal, but we could skip that, one less subscription to invoke for nothing.

The reason why previously `IQueryCollection` is passed to subscribers is because they can decide to subscribe or not based on QS content, a thing that cannot be done at `match` time since match results can be cached and thus should not be affected by QS. So, IMO, `IQueryCollection` is needed, and thus for consistency `RouteData` can be kept as well.

Reply:
Understandable, but in the context of the demo, is this YAGNI? We don't demo the use of the parameters. We ignore them in all cases.

Reply:
Well, in some way you're right, but we "marked" the engine as near to be production ready, and this change make it less, it prevents a legitimate use and it might also raise time wasting questions. Anyway if I accept expression bodied methods everywhere there is no reason to not accept this.